src/AutoComplete.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useCallback } from 'react';
+import Classnames from 'classnames';
+import { Person } from './types/Person';
+import lodash from 'lodash';
+type Props = {
+  peopleData: Person[];
+  onSelected: (person: Person | null) => void;
+  debounceTime?: number;
+};
+export const AutoComplete: React.FC<Props> = ({
+  peopleData,
+  onSelected,
+  debounceTime = 300,
+}) => {
+  const inputRef = React.useRef<HTMLInputElement>(null);
+  const [onFocusActivated, setOnFocusActivated] = React.useState(false);
+  const [input, setInput] = React.useState('');
+  const [filteredPeople, setFilteredPeople] =
+    React.useState<Person[]>(peopleData);
+
+  const [selectedPerson, setSelectedPerson] = React.useState<Person | null>(
+    null,
+  );
+
+  const handleFocus = () => {
+    setOnFocusActivated(true);
+  };
+
+  const handleBlur = () => {
+    setOnFocusActivated(false);
+  };
+
+  const filterPeople = useCallback(
+    lodash.debounce((inputValue: string) => {
+      const filtered = peopleData.filter(person =>
+        person.name.toLowerCase().includes(inputValue.toLowerCase()),
+      );
+
+      setFilteredPeople(filtered);
+    }, debounceTime),
+    [peopleData, input],
```

Comment:
The dependency array for `useCallback` should include `debounceTime` to ensure that the debounce function is updated if `debounceTime` changes.